src/Interpreters/Context.cpp

```diff
@@ -1409,15 +1409,20 @@ void Context::addQueryAccessInfo(
 void Context::addQueryAccessInfo(const Names & partition_names)
 {
     if (isGlobalContext())
-    {
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Global context cannot have query access info");
-    }
 
     std::lock_guard<std::mutex> lock(query_access_info.mutex);
     for (const auto & partition_name : partition_names)
-    {
         query_access_info.partitions.emplace(partition_name);
-    }
+}
+
+void Context::addQueryAccessInfo(const String & qualified_projection_name)
```

Comment:
Maybe `void Context::addQueryAccessInfo(const QualifiedProjectionName & projection_name)` with:
```C++
struct QualifiedProjectionName
{
    String full_table_name;
    String projection_name;
};
```

That would move the caller site a little bit less messy.